src/Connector.sol

```diff
@@ -262,54 +333,42 @@ contract CentrifugeConnector {
     }
 
     function addTranche(
-        uint64 poolId,
-        bytes16 trancheId,
-        string memory tokenName,
-        string memory tokenSymbol,
-        uint8 decimals,
-        uint128 price
+        uint64 _poolId,
+        bytes16 _trancheId,
+        string memory _tokenName,
+        string memory _tokenSymbol,
+        uint8 _decimals,
+        uint128 _price
     ) public onlyGateway {
-        Pool storage pool = pools[poolId];
+        Pool storage pool = pools[_poolId];
         require(pool.createdAt > 0, "CentrifugeConnector/invalid-pool");
 
-        Tranche storage tranche = tranches[poolId][trancheId];
-        require(tranche.lastPriceUpdate == 0, "CentrifugeConnector/tranche-already-added");
-        tranche.latestPrice = price;
-        tranche.lastPriceUpdate = block.timestamp;
-        tranche.tokenName = tokenName;
-        tranche.tokenSymbol = tokenSymbol;
-        tranche.decimals = decimals;
-
-        emit TrancheAdded(poolId, trancheId);
-    }
+        address tranche = tranches[_poolId][_trancheId];
+        require(tranche == address(0), "CentrifugeConnector/tranche-already-added");
+        address asset = address(0x6B175474E89094C44Da98b954EedeAC495271d0F); // TODO FIX : provide tranche currency / assets : default DAI?
+        tranche = deployTranche(poolId, trancheId, asset, tokenName, tokenSymbol, decimals);
```

Comment:
This was moved to a separate permissionless method because calling `AddTranche` over a bridge otherwise had too high of a gas cost.